base/setup/lib/bootsup.c

```diff
@@ -906,37 +811,36 @@ InstallFat16BootCodeToFile(
         return Status;
     }
 
-    /* Adjust bootsector (copy a part of the FAT BPB) */
-    memcpy(&NewBootSector->OemName,
-           &OrigBootSector->OemName,
-           FIELD_OFFSET(FAT_BOOTSECTOR, BootCodeAndData) -
-           FIELD_OFFSET(FAT_BOOTSECTOR, OemName));
+    /*
+     * Copy the disk signature, the reserved fields and
+     * the partition table from the old MBR to the new one.
+     */
+    RtlCopyMemory(&NewBootSector->Signature,
+                  &OrigBootSector->Signature,
+                  sizeof(PARTITION_SECTOR) - offsetof(PARTITION_SECTOR, Signature)
+                    /* Length of partition table */);
 
     /* Free the original boot sector */
     RtlFreeHeap(ProcessHeap, 0, OrigBootSector);
 
-    /* Write new bootsector to DstPath */
-    RtlInitUnicodeString(&Name, DstPath);
+    /* Write new bootsector to RootPath */
+    RtlInitUnicodeString(&Name, RootPath);
 
     InitializeObjectAttributes(&ObjectAttributes,
                                &Name,
                                0,
                                NULL,
                                NULL);
 
-    Status = NtCreateFile(&FileHandle,
-                          GENERIC_WRITE | SYNCHRONIZE,
-                          &ObjectAttributes,
-                          &IoStatusBlock,
-                          NULL,
-                          FILE_ATTRIBUTE_NORMAL,
-                          0,
-                          FILE_OVERWRITE_IF,
-                          FILE_SYNCHRONOUS_IO_NONALERT | FILE_SEQUENTIAL_ONLY,
-                          NULL,
-                          0);
+    Status = NtOpenFile(&FileHandle,
```

Comment:
Due to some reorganization inside this file you're not looking at the same function. This NtOpenFile is inside InstallMbrBootCodeToDiskHelper(), called from InstallMbrBootCodeToDisk(), and to do the comparison you need to check wrt the original InstallMbrBootCodeToDisk() function.
This is NtOpenFile because one opens a disk, so it's expected to "exist"...